website/mailing_list/utils.py

```diff
@@ -0,0 +1,426 @@
+
+        else:
+            create_list(node_id)
+    else:
+        if 'list' in info.keys():
+            delete_list(node_id)
+
+
+###############################################################################
+# Celery Queued tasks
+###############################################################################
+
+@queued_task
+@app.task(bind=True, max_retries=3, default_retry_delay=3 * 60)  # Retry after 3 minutes
+def celery_create_list(*args, **kwargs):
+    create_list(*args, **kwargs)
+
+@queued_task
+@app.task(bind=True, max_retries=3, default_retry_delay=3 * 60)  # Retry after 3 minutes
+def celery_delete_list(*args, **kwargs):
+    delete_list(*args, **kwargs)
+
+@queued_task
+@app.task(bind=True, max_retries=3, default_retry_delay=3 * 60)  # Retry after 3 minutes
+def celery_update_title(*args, **kwargs):
+    update_title(*args, **kwargs)
+
+@queued_task
+@app.task(bind=True, max_retries=3, default_retry_delay=3 * 60)  # Retry after 3 minutes
+def celery_update_single_user_in_list(*args, **kwargs):
+    update_single_user_in_list(*args, **kwargs)
+
+@queued_task
+@app.task(bind=True, max_retries=3, default_retry_delay=3 * 60)  # Retry after 3 minutes
+def celery_remove_user_from_list(*args, **kwargs):
+    remove_user_from_list(*args, **kwargs)
+
+@queued_task
+@app.task(bind=True, max_retries=3, default_retry_delay=3 * 60)  # Retry after 3 minutes
+def celery_update_multiple_users_in_list(*args, **kwargs):
+    update_multiple_users_in_list(*args, **kwargs)
+
+@queued_task
+@app.task(bind=True, max_retries=3, default_retry_delay=3 * 60)  # Retry after 3 minutes
+def celery_full_update(*args, **kwargs):
+    full_update(*args, **kwargs)
+
+
+###############################################################################
+# Signalled Functions
+###############################################################################
+
+@contributor_added.connect
+def subscribe_contributor_to_mailing_list(node, contributor, auth=None):
+    if node.mailing_enabled and contributor.is_active:
+        subscription = node.get_or_create_mailing_list_subscription()
+        subscription.add_user_to_subscription(contributor, 'email_transactional', save=True)
+        celery_update_single_user_in_list(node._id, contributor._id)
+        node.mailing_updated = True
+        node.save()
+
+@contributor_removed.connect
+def unsubscribe_contributor_from_mailing_list(node, user, auth=None):
+    if node.mailing_enabled:
+        subscription = node.get_or_create_mailing_list_subscription()
+    else:
+        from website.models import NotificationSubscription  # avoid circular import
+        subscription = NotificationSubscription.load(to_subscription_key(node._id, 'mailing_list_events'))
+    if subscription:
+        subscription.remove_user_from_subscription(user)
+        node.mailing_updated = True
+        node.save()
+    celery_remove_user_from_list(node._id, user._id)
+
+
+@user_confirmed.connect
+def resubscribe_on_confirm(user):
```

Comment:
Why is this called "resubscribe"? Is an unconfirmed user being added to the mailing list?

If so: What is the intent of subscribing someone twice? (once when added, once when they confirm an account) Could this lead to a situation where someone finds them back on a mailing list after opting out?